File: SocialNetworkAPI/DAL/Models/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace DAL.Models
{
    public class Post
    {
        public int Id { get; set; }
        [MaxLength(150)]
        [Required]
        public string Content { get; set; }
        public User User { get; set; }
        public int UserId { get; set; }
        public DateTime DateOfCreation { get; set; }
        public ICollection<Like> Likes { get; set; }
        public ICollection<Comment> Comments { get; set; } 
    }
}

File: SocialNetwork-SPA/src/app/_model/Post.ts
import { Likers } from './likers';
import { PostComment } from './PostComment';

export class Post {
    id: number;
    content: string;
    userId: number;
    dateOfCreation: Date;
    userKnownAs: string;
    userPhotoUrl: string;
    likers: Likers[];
    comments: PostComment[];
}
